plugins/CoreHome/vue/src/externalLink.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+/**
+ * Add or replace a URL parameter
+ * @param url
+ * @param paramName
+ * @param paramValue
+ */
+function addParameterToUrl(url: string, paramName:string, paramValue:string): string {
+  let returnUrl = url;
+  /* eslint-disable prefer-template */
+  const rx = new RegExp('\\b(' + paramName + '=).*?(&|#|$)');
+
+  // Replace any existing parameter
+  if (returnUrl.search(rx) >= 0) {
+    /* eslint-disable prefer-template */
+    return returnUrl.replace(rx, '$1' + paramValue + '$2');
+  }
+  // Add new parameter
+  returnUrl = returnUrl.replace(/[?#]$/, '');
+  /* eslint-disable prefer-template */
+  return returnUrl + (returnUrl.indexOf('?') > 0 ? '&' : '?') + paramName + '=' + paramValue;
+}
+
+/**
+ * Takes a raw URL and returns an HTML link tag for the URL, if the URL is for a matomo.org
+ * domain then the URL will be modified to include campaign parameters
+ *
+ * @param url              URL to
+ * @param campaignOverride Optional
+ * @param sourceOverride   Optional
+ * @param mediumOverride   Optional
+ */
+export function externalLink(
+  url: string,
+  campaignOverride: string,
+  sourceOverride: string,
+  mediumOverride: string,
+): string {
+  if (!url) {
+    return '';
+  }
+
+  // Check if matomo.org domain
+  const domains = ['matomo.org', 'www.matomo.org', 'developer.matomo.org', 'plugins.matomo.org'];
```

Comment:
Checking for `matomo.org` actually already includes all following checks. So no need for checking them as well...

Reply:
Good point :+1: This array was copied from a similar function in `addCampaignParametersToMatomoLink` which does the check differently, I'll reduce it to just `matomo.org` but keep it as an array so additional domains can easily be added in the future.